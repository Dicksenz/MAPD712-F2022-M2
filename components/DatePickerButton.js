import React from "react";

import { StyleSheet, Text, View, TouchableOpacity } from "react-native";

import Icon from "@expo/vector-icons/MaterialCommunityIcons";

const DatePickerButton = ({ onPress, value }) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.container}>
        <Icon style={{ flex: 1 }} name="calendar" color="grey" size={25} />
        <Text style={{ ...styles.textStyle, flex: 2 }}>{value}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    display: "flex",
    flexDirection: "row",
    alignItems: "center",

    height: 45,
    marginTop: 10,
    marginBottom: 15,
    borderWidth: 1,
    padding: 10,
  },
  textStyle: {
    fontSize: 16,
    color: "grey",
  },
});

export default DatePickerButton;
