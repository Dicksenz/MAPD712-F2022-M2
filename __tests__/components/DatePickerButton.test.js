import React from "react";
import renderer from "react-test-renderer";
import DatePickerButton from "../../components/DatePickerButton";

test("Check if DatePickerButton component renders correctly", () => {
  const tree = renderer.create(<DatePickerButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
