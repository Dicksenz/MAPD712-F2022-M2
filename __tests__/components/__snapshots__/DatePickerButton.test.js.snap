// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if DatePickerButton component renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "height": 45,
        "marginBottom": 15,
        "marginTop": 10,
        "padding": 10,
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "color": "grey",
          "flex": 2,
          "fontSize": 16,
        }
      }
    />
  </View>
</View>
`;
